### ISSUE TITLE
Top Left priority cropping

### ISSUE THREAD

**Issue by VCasecnikovs** [zylo117/Yet-Another-EfficientDet-Pytorch#181]
Hello, 
In class Resizer because of the line: 
`new_image[0:resized_height, 0:resized_width] = image`
If an image is not rectangular, it crops right and down, so at the end image will be shifted to the Top Left corner. Knowing that basically the object stays at the center of the image, it could make some problems to some datasets.
I suggest using some cropper at first, so the Image would always be centered either random cropped.

**Comment by VCasecnikovs** [comment]
One more option would be to use something like this
`[(resized_width - img_size)//2 : resize_width]` and
`[(resized_height - img_size)//2 : resize_height]`
This code can gently do the right crop.

**Comment by VCasecnikovs** [comment]
I forgot that y cropping should also be added.

**Comment by zylo117** [comment]
1. No, there is no cropping during resizing image, only aspectaware resizing. Either right side or down side is padding with zeros.
2. What you do now is my original method, actually. To honor the tradition of the official impl, I switched to this kind of resizing. However, since the pretrained weights is already trained this way, if I alter this now, the mAP will drop.
